Add a dashboard for torchao

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140487

Summary:
trying to add a similar dashboard as https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Test Plan:
Run locally:

install torchao
```
pip install --no-use-pep517 --user "git+https://github.com/pytorch/ao.git@f96e5ec9d30f778c2fb49d7ff9536b8016e1d408"
```
run test:
```
cd benchmarks/torchao
python benchmark.py
```



Check after landing
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Add a dashboard for torchao

Summary:
trying to add a similar dashboard as https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Test Plan:
Check after landing?
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

yanboliang commented on benchmarks/torchao/benchmark.py:
```diff
@@ -0,0 +1,308 @@
+import argparse
```
Is there a strong reason that we need to duplicate this folder? Can we just refer the original one?

jerryzh168 replied:
what do you mean by original one?

jerryzh168 replied:
Sounds good, I can use gpt-fast

jerryzh168 committed:
Update on "Add a dashboard for torchao"


Summary:
trying to add a similar dashboard as https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Test Plan:
Check after landing?
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Add a dashboard for torchao"


Summary:
trying to add a similar dashboard as https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Test Plan:
Check after landing?
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Add a dashboard for torchao"


Summary:
trying to add a similar dashboard as https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Test Plan:
Run locally:

install torchao
```
pip install --no-use-pep517 --user "git+https://github.com/pytorch/ao.gitf96e5ec9d30f778c2fb49d7ff9536b8016e1d408"
```
run test:
```
cd benchmarks/torchao
python benchmark.py
```



Check after landing
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
will be using https://github.com/pytorch/pytorch/pull/140627 instead
